lib/Client.php

```diff
@@ -66,19 +66,31 @@ class Client extends EventEmitter
      */
     protected $maxRedirects = 5;
 
+    protected $headerLines = [];
+
     /**
      * Initializes the client.
      */
     public function __construct()
     {
         $this->curlSettings = [
             CURLOPT_RETURNTRANSFER => true,
-            CURLOPT_HEADER => true,
+            CURLOPT_HEADERFUNCTION => [$this, 'receiveCurlHeader'],
             CURLOPT_NOBODY => false,
             CURLOPT_USERAGENT => 'sabre-http/'.Version::VERSION.' (http://sabre.io/)',
         ];
     }
 
+    protected function receiveCurlHeader($curlHandle, $headerLine)
+    {
+        if ($this->curlHandle !== $curlHandle) {
```

Comment:
Can this actually occur?

Reply:
I have removed this line and revised the changes to take `sendAsync` into  consideration.